cppcache/src/ExceptionTypes.cpp

```diff
@@ -23,30 +23,61 @@
 #include "ErrType.hpp"
 #include "util/Log.hpp"
 
-namespace apache {
-namespace geode {
-namespace client {
-
-void setThreadLocalExceptionMessage(std::string exMsg);
-const std::string& getThreadLocalExceptionMessage();
-
-static std::map<GfErrType, std::function<void(std::string, std::string&,
-                                              GfErrType, std::string)>>
-    error_map;
+namespace {
+
```

Comment:
Wait, why did you pull this out of the namespace and then import all these exceptions?

Reply:
I am glad you noticed. There is a subtle, still pretty good reason for this.
I've made a gist demonstrating my point: https://gist.github.com/gaussianrecurrence/cda4f6c16b2652e09b539ae9c2bf6e92

Summarizing, basically there is a non-written rule that states: "If you implement free functions that are to be used only within the module implementing them, then put them in a anonymous namespace"

That's done in order to avoid naming conflicts.